fundamentals/fundamentals-bio/src/test/scala/izumi/fundamentals/bio/MiniBIOCatsLaws.scala

```diff
@@ -1,72 +1,72 @@
-//      case Right(b) => pure(b)
-//    }
-//
-//    override def pure[A](x: A): MiniBIO[Throwable, A] = bio.now(x)
-//  }
-//
-//  val tests = new SyncTests[MiniBIO[Throwable, ?]] {
-//    val laws = new SyncLaws[MiniBIO[Throwable, ?]] {
-//      val F = Sync
-//    }
-//  }
-//
-//  implicit def arb[A](implicit arb: Arbitrary[A]): Arbitrary[MiniBIO[Throwable, A]] = Arbitrary {
-//    Arbitrary.arbBool.arbitrary.flatMap {
-//      if (_)
-//        arb.arbitrary.map(BIO[MiniBIO].now(_))
-//      else
-//        Arbitrary.arbThrowable.arbitrary.map(BIO[MiniBIO].fail(_))
-//    }
-//  }
-//
-//  implicit def eq[A](implicit eq: Eq[A]): Eq[MiniBIO[Throwable, A]] = Eq.instance {
-//    (l, r) => l.run() == r.run()
-//  }
-//
-//  checkAll("Sync[MiniBIO]", tests.sync[Int, Int, Int])
-//
-//}
+class MiniBIOCatsLaws extends AnyFunSuite with FunSuiteDiscipline with TestInstances with Checkers {
+
+  val bio = BIO[MiniBIO]
+  implicit val Sync: Sync[MiniBIO[Throwable, ?]] = new Sync[MiniBIO[Throwable, ?]] {
+    override def suspend[A](thunk: => MiniBIO[Throwable, A]): MiniBIO[Throwable, A] = bio.syncThrowable(thunk).flatten
+
+    override def bracketCase[A, B](
+      acquire: MiniBIO[Throwable, A]
+    )(use: A => MiniBIO[Throwable, B]
+    )(release: (A, ExitCase[Throwable]) => MiniBIO[Throwable, Unit]
+    ): MiniBIO[Throwable, B] = {
+      bio.bracketCase[Any, Throwable, A, B](acquire)(
+        (a, e) =>
+          release(
+            a,
+            e match {
+              case BIOExit.Success(_) => ExitCase.Completed
+              case value: BIOExit.Failure[Throwable] => ExitCase.Error(value.toThrowable)
+            },
+          ).orTerminate
+      )(use)
+    }
+
+    override def raiseError[A](e: Throwable): MiniBIO[Throwable, A] = bio.fail(e)
+
+    override def handleErrorWith[A](fa: MiniBIO[Throwable, A])(f: Throwable => MiniBIO[Throwable, A]): MiniBIO[Throwable, A] = fa.catchAll(f)
+
+    override def flatMap[A, B](fa: MiniBIO[Throwable, A])(f: A => MiniBIO[Throwable, B]): MiniBIO[Throwable, B] = fa.flatMap(f)
+
+    override def tailRecM[A, B](a: A)(f: A => MiniBIO[Throwable, Either[A, B]]): MiniBIO[Throwable, B] = f(a).flatMap {
+      case Left(a) => tailRecM(a)(f)
+      case Right(b) => pure(b)
+    }
+
+    override def pure[A](x: A): MiniBIO[Throwable, A] = bio.pure(x)
+  }
+
+  val tests = new SyncTests[MiniBIO[Throwable, ?]] {
+    val laws = new SyncLaws[MiniBIO[Throwable, ?]] {
+      val F = Sync
+    }
+  }
+
+  implicit def arb[A](implicit arb: Arbitrary[A]): Arbitrary[MiniBIO[Throwable, A]] = Arbitrary {
+    Arbitrary.arbBool.arbitrary.flatMap {
+      if (_)
+        arb.arbitrary.map(BIO[MiniBIO].pure(_))
+      else
+        Arbitrary.arbThrowable.arbitrary.map(BIO[MiniBIO].fail(_))
+    }
+  }
+
+  implicit def eq[A](implicit eq: Eq[A]): Eq[MiniBIO[Throwable, A]] = Eq.instance {
+    (l, r) => l.run() == r.run()
+  }
+
+  checkAll("Sync[MiniBIO]", tests.sync[Int, Int, Int])
```

Comment:
Now we can also check `ConcurrentTests`/`ConcurrentLaws` for `zio.IO` & `monix.bio.IO`

Reply:
Okay, I'll implement those too.

@neko-kai is it OK that MiniBIO does not corresponds to all SyncLaws?

Reply:
@VladPodilnyk That means there's an error in MiniBIO most likely, but you may ignore it for now. I may look at it in a separate PR.

Reply:
Okay, got it.

Reply:
@neko-kai, I don't think that it's a bug in MiniBIO. I have suspicions that the  problem is the `Eq` typeclass we provide for MiniBIO and others. I've pushed a commit where I've tested ConcurrentLaws. Both ZIO and Monix haven't passed a test.

Reply:
@VladPodilnyk Could you check with `Eq` from `zio-interop-cats` - https://github.com/zio/interop-cats/blob/master/interop-cats/shared/src/test/scala/zio/interop/catzSpecBase.scala?

Reply:
@VladPodilnyk which tests fail exactly?

Reply:
Small part of what laws fails/pass for monix
<img width="498" alt="image" src="https://user-images.githubusercontent.com/12114216/94190548-ec490c00-feb4-11ea-9462-d101a33b6141.png">

Same for ZIO
<img width="483" alt="image" src="https://user-images.githubusercontent.com/12114216/94190616-04b92680-feb5-11ea-8ad2-7003750c9abd.png">

Reply:
@VladPodilnyk You can progress on the issue by checking if `zio-interop-cats` instances pass the tests, and if not – narrowing down the difference between `zio-interop-cats` test suite / test environment and ours.

Reply:
@VladPodilnyk You can checkout and play with https://github.com/zio/interop-cats/ project and try to detect the differences

Reply:
thanks for suggestions, I'll try!